gradio/themes/base.py

```diff
@@ -32,7 +46,179 @@ def repl_func(match):
         dark_css += "}"
         return css + "\n" + dark_css
 
+    def to_dict(self):
+        """Convert the theme into a python dictionary."""
+        schema = {"theme": {}}
+        for prop in dir(self):
+            if not prop.startswith("_") and isinstance(getattr(self, prop), str):
+                schema["theme"][prop] = getattr(self, prop)
+        return schema
+
+    @classmethod
+    def load(cls, path: str) -> "ThemeClass":
+        """Load a theme from a json file.
+
+        Parameters:
+            path: The filepath to read.
+        """
+        theme = json.load(open(path))
+        return cls.from_dict(theme)
+
+    @classmethod
+    def from_dict(cls, theme: Dict[str, Dict[str, str]]) -> "ThemeClass":
+        """Create a theme instance from a dictionary representation.
+
+        Parameters:
+            theme: The dictionary representation of the theme.
+        """
+        base = cls()
+        for prop, value in theme["theme"].items():
+            setattr(base, prop, value)
+        return base
+
+    def dump(self, filename: str):
+        """Write the theme to a json file.
+
+        Parameters:
+            filename: The path to write the theme too
+        """
+        as_dict = self.to_dict()
+        json.dump(as_dict, open(Path(filename), "w"))
+
+    @classmethod
+    def from_hub(cls, repo_name: str):
+        """Load a theme from the hub.
+
+        This DOES NOT require a HuggingFace account.
+
+        Parameters:
+            repo_name: string of the form <author>/<theme-name>@<semantic-version-expression>.  If a semantic version expression is omitted, the latest version will be fetched.
+        """
+        if "@" not in repo_name:
+            name, version = repo_name, None
+        else:
+            name, version = repo_name.split("@")
+
+        assets = get_theme_assets(name)
+        matching_version = get_matching_version(assets, version)
+
+        if not matching_version:
+            raise ValueError(
+                f"Cannot find a matching version for expression {version} "
+                f"from files {[f.filename for f in assets]}"
+            )
+
+        url = huggingface_hub.hf_hub_url(
+            repo_id=name,
+            repo_type="space",
+            filename=f"themes/theme_schema@{matching_version.version}.json",
+        )
+        download = requests.get(url)
+        if download.ok:
+            return cls.from_dict(download.json())
+        else:
+            raise ValueError(
+                f"Error downloading vile themes/theme_schema@{matching_version}.json from {repo_name}!"
+            )
```

Comment:
Perhaps we should fallback to `DefaultTheme()` if the theme cannot be found and raise a warning instead of an exception? I'm thinking about the case where someone is using someone else's theme and the theme gets deleted. It would be good if the Space is still runnable

Reply:
I think that would be more confusing for users using `ThemeClass.from_hub` actually. I see your point though.

How about this, `from_hub` should raise an error if the theme can't be found but when passing a string-valued theme to `Blocks` or `Interface`, this error should be caught and the `DefaultTheme` should be used instead?

Reply:
I think that's a happy medium. App authors who just want to load a theme are likely to use the `Blocks(theme="abidlabs/....")` syntax. Theme creators who are loading a theme and modifying it will use `from_hub` method instead.

Reply:
Yeah sounds good!